packages/common.vm/tools/vm.common/vm.common.psm1

```diff
@@ -496,6 +496,101 @@ function VM-Install-From-Zip {
+    [Parameter(Mandatory=$true, Position=2)]
+    [string] $exeUrl,
+    [Parameter(Mandatory=$false)]
+    [string] $exeSha256,
+    [Parameter(Mandatory=$false)]
+    [string] $exeUrl_64,
+    [Parameter(Mandatory=$false)]
+    [string] $exeSha256_64
+  )
+  try {
+    $toolDir = Join-Path ${Env:RAW_TOOLS_DIR} $toolName
+    $shortcutDir = Join-Path ${Env:TOOL_LIST_DIR} $category
+
+    # Download and install
+    $executablePath = Join-Path $toolDir ($toolName + ".exe")
+    $packageArgs = @{
+      packageName = ${Env:ChocolateyPackageName}
+      url = $exeUrl
+      checksum = $exeSha256
+      checksumType = "sha256"
+      fileFullPath = $executablePath
+      forceDownload = $true
+    }
+    Get-ChocolateyWebFile @packageArgs
+    VM-Assert-Path $executablePath
+
+    $executableCmd  = Join-Path ${Env:WinDir} "system32\cmd.exe" -Resolve
+    $executableDir  = Join-Path ${Env:UserProfile} "Desktop" -Resolve
+    $executableArgs = "/K `"cd `"$executableDir`" && `"$executablePath`" --help`""
+    $shortcut = Join-Path $shortcutDir ($toolName + ".lnk")
+    Install-ChocolateyShortcut -shortcutFilePath $shortcut -targetPath $executableCmd -Arguments $executableArgs -WorkingDirectory $executableDir -RunAsAdmin
+    VM-Assert-Path $shortcut
+    Install-BinFile -Name $toolName -Path $executablePath
+  } catch {
+    VM-Write-Log-Exception $_
+  }
+}
+
+function VM-Install-Single-Ps1 {
+  [CmdletBinding()]
+  Param
+  (
+    [Parameter(Mandatory=$true, Position=0)]
+    [string] $toolName,
+    [Parameter(Mandatory=$true, Position=1)]
+    [string] $category,
+    [Parameter(Mandatory=$true, Position=2)]
+    [string] $ps1Url,
+    [Parameter(Mandatory=$false)]
+    [string] $ps1Sha256,
+    [Parameter(Mandatory=$false)]
+    [string] $ps1Url_64,
+    [Parameter(Mandatory=$false)]
+    [string] $ps1Sha256_64
+  )
+  try {
+    $toolDir = Join-Path ${Env:RAW_TOOLS_DIR} $toolName
+    $shortcutDir = Join-Path ${Env:TOOL_LIST_DIR} $category
+
+    # Download and install
+    $scriptPath = Join-Path $toolDir ($toolName + ".ps1")
+    $packageArgs = @{
+      packageName = ${Env:ChocolateyPackageName}
+      url = $ps1Url
+      checksum = $ps1Sha256
+      checksumType = "sha256"
+      fileFullPath = $scriptPath
+      forceDownload = $true
+    }
+    Get-ChocolateyWebFile @packageArgs
+    VM-Assert-Path $scriptPath
+    
+    # Create shortcut
+    $target_cmd = Join-Path ${Env:WinDir} "system32\cmd.exe" -Resolve
+    $target_args = '/K powershell.exe -ExecutionPolicy Bypass -NoExit -Command "cd ' + $toolDir + '"'
+    $target_dir = $toolDir
+    $target_icon = Join-Path (Join-Path ${Env:WinDir} "system32\WindowsPowerShell\v1.0") "powershell.exe" -Resolve
+
+    $shortcut = Join-Path $shortcutDir ($toolName + ".ps1.lnk")
+    Install-ChocolateyShortcut -shortcutFilePath $shortcut -targetPath $target_cmd -Arguments $target_args -WorkingDirectory $target_dir -IconLocation $target_icon -RunAsAdmin
+    VM-Assert-Path $shortcut
+
```

Comment:
Allow users to specify an additional PowerShell command to execute after changing into the script directory. For instance: `Import-Module .\WMIOps.ps1; Get-Content README.md`. Used the `$ps1Cmd` here to allow this.

I also tried to consider paths that have spaces in them and properly quote -- initial testing seems good, but we should test later.

```suggestion
    $shortcut = Join-Path $shortcutDir "$toolName.ps1.lnk"
    $targetCmd = Join-Path ${Env:WinDir} "system32\cmd.exe" -Resolve
    if ($ps1Cmd) {
      $targetArgs = "/K powershell.exe -ExecutionPolicy Bypass -NoExit -Command `"cd '$toolDir'; $ps1Cmd`""
    } else {
      $targetArgs = "/K powershell.exe -ExecutionPolicy Bypass -NoExit -Command `"cd '$toolDir'`""
    }
    $targetIcon = Join-Path (Join-Path ${Env:WinDir} "system32\WindowsPowerShell\v1.0") "powershell.exe" -Resolve
    
    Install-ChocolateyShortcut -shortcutFilePath $shortcut -targetPath $targetCmd -arguments $targetArgs -workingDirectory $toolDir -iconLocation $targetIcon
    VM-Assert-Path $shortcut

```